### PR TITLE
Check environment variables for appengine

### PR DESCRIPTION
**Pull request by mattwhisenhunt**
@lesv PTAL

### CODE REVIEW

**Comment by lesv** [comment]
Let me make sure this is true on Java7 and then I'll ok.

**Comment by lesv** [comment]
The change you've made will identify GAE/STD (java 7, Java 8) and GAE Flex (compat).   I still need to check non-compat, I think it might which will be a change in behavior.

**Comment by lesv** [comment]
Java 8 on GAE Standard:


DATACENTER | us10
-- | --
GAE_DEPLOYMENT_ID | 402856547336086879
GAE_APPLICATION | s~XXX-003
GOOGLE_CLOUD_PROJECT | XXX-003
GAE_SERVICE | XXXX
GAE_ENV | standard
GAE_INSTANCE | 00c61b117c4997505a125e9a7aa139bffd4f3a1f32420029bec3c86d6239c3c935b3e89bd30a4a
USER | appengine
TMPDIR | /tmp
GCLOUD_PROJECT | XXX-003
GAE_VERSION | XXXX:20170722t140621
GAE_RUNTIME | java8

And Java 7 has:

GAE_ENV | standard
-- | --
GAE_RUNTIME | java7

Oops - I gave bad advice above.  First check for GAE_ENV or GAE_RUNTIME (that's GAE standard), and then check for GAE_VM (GAE Flex Compat).  If it's a normal Flex runtime, you probably want to use the normal GCP stuff.

**Comment by lesv** [comment]
I've checked w/ Eng and my final comments above appear to be correct.

**Comment by mattwhisenhunt** [comment]
So if we are running on Flex/Compat, runningOnAppEngine() should return false? Is that right?

**Comment by lesv** [comment]
Felx/Compat RunningOnAppEngine() probably should be TRUE for most services.
Flex/non-compat - should be false

**Comment by mattwhisenhunt** [comment]
@lesv PTAL

File: google-api-client/src/main/java/com/google/api/client/googleapis/auth/oauth2/DefaultCredentialProvider.java [hunk 1]
```diff
@@ -259,38 +258,18 @@ private GoogleCredential getCredentialUsingWellKnownFile(
...
 
   private boolean runningOnAppEngine() {
-    Class<?> systemPropertyClass = null;
-    try {
-      systemPropertyClass = forName("com.google.appengine.api.utils.SystemProperty");
-    } catch (ClassNotFoundException expected) {
-      // SystemProperty will always be present on App Engine.
-      return false;
+    Map<String, String> env = System.getenv();
+    if (env.containsKey("GAE_ENV") && env.get("GAE_VM").equals("standard")) {
```
Comment (lesv) [inline google-api-client/src/main/java/com/google/api/client/googleapis/auth/oauth2/DefaultCredentialProvider.java]: I don't think GAE_VM ever equals "standard"

Comment (lesv) [reply google-api-client/src/main/java/com/google/api/client/googleapis/auth/oauth2/DefaultCredentialProvider.java]: I think you meant to check `GAE_ENV`, not `GAE_VM`

**Comment by lesv** [comment]
Also, you might wish to see why TRAVIS is failing.

File: google-api-client/src/main/java/com/google/api/client/googleapis/auth/oauth2/DefaultCredentialProvider.java (hunk 1 above)
Comment (mattwhisenhunt) [reply google-api-client/src/main/java/com/google/api/client/googleapis/auth/oauth2/DefaultCredentialProvider.java]: Indeed, copy and paste error… Thanks

**Comment by mattwhisenhunt** [comment]
Fixes #1083

**Comment by mattwhisenhunt** [comment]
@lesv PTAL

**Comment by lesv** [comment]
Travis is unhappy that @mattwhisenhunt is not a member of committers.  I'm going to merge.

**Pull request merged**